### PR TITLE
PT-156728559 Cleanup requests, PeerId and convert hostname to binary

### PR DESCRIPTION
**Pull request by hanssv**
[x] When we drop the TCP connection error (and remove) all outstanding requests
[x] `aec_peers` had hostname and port in its PeerId - only PubKey is necessary
[x] `inet:parse/1` returns a `string()` it should be converted to a binary.

### CODE REVIEW

File: apps/aecore/src/aec_peers.erl [hunk 1]
```diff
@@ -149,7 +147,7 @@ remove(Peer) ->
 %%------------------------------------------------------------------------------
 %% Get list of all peers. The list may be big. Use with caution.
 %% Consider using get_random instead.
-%%------------------------------------------------------------------------------
+%------------------------------------------------------------------------------
```
Comment (gorillainduction) [inline apps/aecore/src/aec_peers.erl]: Cosmetic bug ;-)

Comment (uwiger) [inline apps/aecore/src/aec_peers.erl]: Why?

Comment (hanssv) [reply apps/aecore/src/aec_peers.erl]: New keyboard, I sometimes the wrong keys which gives interesting effects in vi/vim...

Comment (hanssv) [reply apps/aecore/src/aec_peers.erl]: New keyboard, I sometimes the wrong keys which gives interesting effects in vi/vim...

**Pull request merged**